src/lib/rest/OrionError.cpp

```diff
@@ -144,62 +151,33 @@ std::string OrionError::render(ConnectionInfo* ciP, const std::string& _indent)
 }
 
 
+
 /* ****************************************************************************
 *
 * OrionError::render -
 *
-* FIXME P3: OrionError() render method should be unified (probably the class itself needs
-* a good refactoring, to live in the NGSIv1-NGSIv2 dual world...)
+* This method removes any whitespace in the reasonPhrase field, i.e.
+* transforms "Not Found" to "NotFound".
 *
-* Simplified version, without ciP
+* It is used by smartRender method, in order to prepare to render in API v2 case
 *
 */
-std::string OrionError::render(const std::string& _indent, const std::string apiVersion)
+void OrionError::shrinkReasonPhrase(void)
 {
-  //
-  // For API version 2 this is pretty easy ...
-  //
-  if (apiVersion == "v2")
-  {
-    if (details == "Already Exists")
-    {
-      details = "Entity already exists";
-    }
+  char buf[80];  // 80 should be enough to hold any reason phrase
 
-    reasonPhrase = errorStringForV2(reasonPhrase);
-    return "{" + JSON_STR("error") + ":" + JSON_STR(reasonPhrase) + "," + JSON_STR("description") + ":" + JSON_STR(details) + "}";
-  }
+  strncpy(buf, reasonPhrase.c_str(), 80);
```

Comment:
80 => sizeof(buf)

Using sizeof, all is still ok if the size of buf changes, avoiding a possible bug in the future.